core_types/src/entry/entry_type.rs

```diff
@@ -14,17 +14,18 @@ macro_rules! sys_prefix {
 // Variant `Data` is for user defined entry types
 #[derive(Debug, Clone, PartialEq, Hash, Serialize, Deserialize)]
 pub enum EntryType {
-    AgentId,
-    Deletion,
-    App(String),
     Dna,
+
+    Agent,
+
+    App(String),
+    Delete,
+
+    LinkAdd,
+    LinkRemove,
+
     ChainHeader,
-    Key,
-    Link,
-    Migration,
-    /// TODO #339 - This is different kind of SystemEntry for the DHT only.
-    /// Should be moved into a different enum for DHT entry types.
-    LinkList,
+    ChainMigrate,
 }
```

Comment:
How about separating this into two enums:

```Rust
enum EntryType {
 App(String),
 System(SystemEntryType),
}

enum SystemEntryType {
 AgentId,
 Dna,
 Link,
...
}
```

Because I believe there will be many places where we handle a SystemEntryType and don't want to make sure its not an AppEntry everytime.

Reply:
@ddd-mtl yes we're doing an enum based approach :)